### ISSUE TITLE
Slow startup

### ISSUE THREAD

**Issue by chillypeppe** [ChristianChiarulli/LunarVim#452]
I just tried LunarVim and I noticed that the startup is quite slow. On my laptop nvim takes twice the time to start than it did with my last configuration (using vimscript and other plugins as coc-nvim).
With laptop on battery I get:
```
407.478  002.648: first screen update
407.482  000.004: --- NVIM STARTED --- 
```
On power supply:
```
211.986  001.671: first screen update
211.989  000.002: --- NVIM STARTED --- 
```
With my previous config nvim took respectively 210ms and 120ms to start.

I noticed that the elapsed time for sourcing vimrc files is A LOT:
115ms on battery
```
163.013  115.652: sourcing vimrc file(s)
```
and 217 on power supply
```
338.756  217.568: sourcing vimrc file(s)
```

I can't find the cause of the massive delay. Disabling one plugin at a time didn't help.

**Comment by rebuilt** [comment]
There are a lot of files to read and a lot of custom configuration set.  (80 files)  Also, every plugin gets loaded at start.  

If you're curious you can check out this commit which improves startup time
```bash
git fetch origin pull/416/head:startup
git checkout startup
```
To switch back to normal lunarvim
```bash
git checkout master
```
I think the code in this pull request was overwritten by later commits but I'm not sure.  

Out of curiosity does your laptop use a traditional spinning disk drive or does it have an ssd?

**Comment by chillypeppe** [comment]
This definitely speeds things up, thank you a lot. I figured it might have been that every plugin got loaded at startup, but I'm still learning lua and init.lua so I didn't know how to do it.
 
Besides I made a few tests and I would advise to load at least nvim-compe at startup because the little freeze after you enter insert mode is a bit annoying and loading it at startup doesn't seem to bring a big performance loss.

These are the startup speeds after using that commit (with nvim-compe immediately loaded):
On battery:
```
243.263  001.396: first screen update
243.267  000.004: --- NVIM STARTED ---
```
On power supply:
```
120.166  001.020: first screen update
120.168  000.002: --- NVIM STARTED ---
```

To answer to your last question, I have an ssd but I do have full disk encryption, so that might slow things up a bit.
